code/game/data_huds.dm

```diff
@@ -97,12 +97,23 @@
 	var/image/onlineholder = hud_list[ONLINE_HUD]
 	var/icon/I = icon(icon, icon_state, dir)
 	onlineholder.pixel_y = I.Height() - world.icon_size
-	if(!istype(src, /mob/living/carbon/human) && !client)
-		onlineholder.icon_state = "none"
-	else if(istype(src, /mob/living/carbon/human) && !client || !key)
+	if(!client || !key)
 		onlineholder.icon_state = "offline"
 	else
-		onlineholder.icon_state = "online"
+		onlineholder.icon_state = "none"
+
+/mob/living/carbon/human/Logout()
+	..()
+	hud_client_check()
+
+/mob/living/carbon/human/Login()
+	..()
+	hud_client_check()
+
+/mob/living/carbon/human/Initialize()
+	..()
+	var/datum/atom_hud/data/human/client/clienthud = GLOB.huds[DATA_HUD_CLIENT]
+	clienthud.add_hud_to(src)
```

Comment:
This will cause issues.
Always preserve the parent return, and don't re-define an existent proc again without very good reasons.